Lib/test/test_hashlib.py

```diff
@@ -901,36 +901,17 @@ def test_get_fips_mode(self):
         if fips_mode is not None:
             self.assertIsInstance(fips_mode, int)
 
-    def test_disallow_instanciation(self):
-        constructors = []
-        try:
-            import _md5
-            constructors.append(_md5.md5)
-        except ImportError:
-            pass
-        try:
-            import _sha1
-            constructors.append(_sha1.sha1)
-        except ImportError:
-            pass
-        try:
-            import _sha256
-            constructors.append(_sha256.sha224)
-            constructors.append(_sha256.sha256)
-        except ImportError:
-            pass
-        try:
-            import _sha512
-            constructors.append(_sha512.sha384)
-            constructors.append(_sha512.sha512)
-        except ImportError:
-            pass
-
-        for constructor in constructors:
-            h = constructor()
-            with self.subTest(constructor=constructor):
-                hash_type = type(h)
-                self.assertRaises(TypeError, hash_type)
+    def test_disallow_instantiation(self):
```

Comment:
Perhaps add `test.support.cpython_only` decorator?

Reply:
I assumed Victor didn't add the decorator for a reason.

Reply:
I'm guessing he forgot :) See Serhiy's [msg392169](https://bugs.python.org/issue43916#msg392169) in the bpo. Ditto for GH-24482.

Reply:
Thanks for the update!